test/FormFileSpec.js

```diff
@@ -0,0 +1,80 @@
+import { mount } from 'enzyme';
+import React from 'react';
+import FormFile from '../src/FormFile';
+
+describe('<FormFile>', () => {
+  it('should render correctly', () => {
+    let wrapper = mount(
+      <FormFile id="foo" name="foo" label="My label" className="my-file" />,
+    );
+
+    wrapper
+      .assertSingle('div.form-group.my-file')
+      .assertSingle('input[type="file"][name="foo"]');
+
+    wrapper
+      .assertSingle('label.form-file-label[htmlFor="foo"]')
+      .text()
+      .should.equal('My label');
+  });
+
+  it('should support isValid', () => {
+    mount(<FormFile isValid />).assertSingle('.is-valid');
+  });
+
+  it('should support isInvalid', () => {
+    mount(<FormFile isInvalid />).assertSingle('.is-invalid');
+  });
+
+  it('should support ref forwarding', () => {
+    class Container extends React.Component {
+      render() {
+        return (
+          <FormFile
+            ref={ref => {
+              this.input = ref;
+            }}
+          />
+        );
+      }
+    }
+
+    const instance = mount(<Container />).instance();
+
+    expect(instance.input.tagName).to.equal('INPUT');
+  });
+
+  it('should supports custom', () => {
+    const wrapper = mount(<FormFile custom label="My label" />);
+
+    wrapper
+      .assertSingle('div.custom-file')
+      .assertSingle('input.custom-file-input');
+
+    wrapper.assertSingle('label.custom-file-label');
+  });
+
+  it('should supports lang when custom', () => {
```

Comment:
Is there any way to test with enzyme that the CSS that *should* match based on the lang tag being there?

Reply:
I can answer this myself. EnzymeJS does not currently support testing ::after pseudo-elements. There is an open issue over at their GitHub: https://github.com/enzymejs/enzyme/issues/2102